lib/list.gi

```diff
@@ -2434,37 +2434,12 @@ end );
 InstallMethod( SortingPerm,
     [ IsDenseList ],
     function( list )
-    local  both, perm, i, l;
+        local copy;
 
-    # {\GAP} supports permutations only up to `MAX_SIZE_LIST_INTERNAL'.
-    if not IsSmallList( list ) then
-      Error( "<list> must have length at most ", MAX_SIZE_LIST_INTERNAL );
-    fi;
-
-    # make a new list that contains the elements of <list> and their indices
-    both := [];
-    l:= Length( list );
-    for i in [ 1 .. l ] do
-        both[i] := [ list[i], i ];
-    od;
-
-    # Sort the new list according to the first item (stable).
-    # This needs more memory than a call of 'Sort' but is much faster.
-    # (The change was proposed by Frank Lübeck.)
-    both := Set( both );
-
-    # Remember the permutation.
-    perm := [];
-    perm{ [ 1 .. l ] }:= both{ [ 1 .. l ] }[2];
-
-    # return the permutation mapping <list> onto the sorted list
-    return PermList( perm )^(-1);
+        copy := ShallowCopy(list);
+        return Sortex(copy);
     end );
 
-InstallMethod( SortingPerm,
-    "for a dense and sorted list",
-    [ IsDenseList and IsSortedList ], SUM_FLAGS,
-    list -> () );
```

Comment:
@ChrisJefferson any particular reason this method was removed? I see no mention of this change in the PR description nor the commit message.

Reply:
It was part of making SortingPerm be exactly the same as SortEx.

Reply:
I don't see why that requires removing this optimization, though? Now if the input is sorted, we make a copy, sort that, and compute the identity permutation from that...

Reply:
It could probably have been left in, I wouldn't have a problem with it being re-added but I wasn't sure it was worth it, for a fairly unusual case.

Reply:
I see this method as an illustration of the idea to avoid working by using known information.
If the presence of such *methods for fairly unusual cases* does not slow down
the computations in the usual cases
then these methods are beneficial.
If this is not the case then it might be interesting to discuss the situation in detail,
perhaps in some tutorial.